docs/machine-learning/how-does-mldotnet-work.md

```diff
@@ -0,0 +1,262 @@
+        {
+            new HouseData() { Size = 1.1F, Price = 0.98F },
+            new HouseData() { Size = 1.9F, Price = 2.1F },
+            new HouseData() { Size = 2.8F, Price = 2.9F },
+            new HouseData() { Size = 3.4F, Price = 3.6F }
+        };
+
+        var testHouseDataView = mlContext.Data.LoadFromEnumerable(houseData);
+        var testPriceDataView = model.Transform(testHouseDataView);
+                
+        var metrics = mlContext.Regression.Evaluate(testPriceDataView, labelColumnName: "Price");
+
+        Console.WriteLine($"R^2: {metrics.RSquared:0.##}");
+        Console.WriteLine($"RMS error: {metrics.RootMeanSquaredError:0.##}");
+
+        // R^2: 0.99
+        // RMS error: 0.09
+```
+
+The evaluation metrics tell us that the error is low-ish, and that correlation between the predicted output and the test output is high. That was easy! In real examples, it takes more tuning to achieve good model metrics.
+
+## Architecture
+
+In this section, we go through the architectural patterns of ML.NET. If you are an experienced .NET developer, some of these patterns will be familiar to you, and some will be less familiar. Hold tight, while we dive in!
+
+An ML.NET application starts with an <xref:Microsoft.ML.MLContext> object. This singleton object contains properties, which are **catalogs** for:
+
+- data loading and saving: <xref:Microsoft.ML.Data>
+- data preparation :<xref:Microsoft.ML.TransformsCatalog>
+- algorithms for model training
+    - binary classification model training: <xref:Microsoft.ML.BinaryClassificationCatalog>
+    - multiclass classification model training: <xref:Microsoft.ML.MulticlassClassificationCatalog>
+    - anomaly detection model training: <xref:Microsoft.ML.AnomalyDetectionCatalog>
+    - ranking model training: <xref:Microsoft.ML.RankingCatalog>
+    - regression model training: <xref:Microsoft.ML.RegressionCatalog>
+- model loading, saving, and creating prediction engines: <xref:Microsoft.ML.ModelOperationsCatalog>
+
+The catalogs are a convenience to help you navigate the available operations in each of the above categories. Using Visual Studio, the catalogs show up via IntelliSense. 
+
+   ![Intellisense for Regression Trainers](./media/catalog-intellisense.png)
+
+### Extension functions
+
+Inside each catalog is a set of extension functions. Let's look at how extension functions are used to create a training pipeline.
+
+```csharp
+    var pipeline = mlContext.Transforms.Concatenate("Features", new[] { "Size" })
+        .Append(mlContext.Regression.Trainers.Sdca(labelColumnName: "Price", maximumNumberOfIterations: 100);
+```
+
+In the snippet, `Concatenate` and `Sdca` are both extension functions. They each create an <xref:Microsoft.ML.IEstimator`1> object that is appended to the pipeline. At this point, the objects are created only. No execution has happened.
+
+### Fit the model
+
+Once the objects in the pipeline have been created, data can be used to train or fit the model.
```

Comment:
Would this be better? "Once the objects in the pipeline have been created, data can be used to fit (train) the model."

Reply:
Removed fit altogether